src/common/netif.cpp

```diff
@@ -276,9 +269,46 @@ std::vector<CNetAddr> GetLocalAddresses()
 {
     std::vector<CNetAddr> addresses;
 #ifdef WIN32
-    char pszHostName[256] = "";
-    if (gethostname(pszHostName, sizeof(pszHostName)) != SOCKET_ERROR) {
-        addresses = LookupHost(pszHostName, 0, true);
+    DWORD status = 0;
+    constexpr size_t MAX_ADAPTER_ADDR_SIZE = 4 * 1000 * 1000; // Absolute maximum size of adapter addresses structure we're willing to handle, as a precaution.
```

Comment:
Added an upper limit here to avoid hanging forever or consuming absurd amounts of resources in case of a windows bug. i would imagine 4MB of adapter addresses ought to be enough for anyone.